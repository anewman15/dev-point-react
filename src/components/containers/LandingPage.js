import { Link } from 'react-router-dom';

function LandingPage() {
  return (
    <div className="bg-code">
      <div className="columns is-centered">
        <div className="my-6 column is-four-fifths">
          <div className="p-6 bd-filter border-warning has-text-white">
            <p className="is-size-2 has-text-centered">
              Welcome to DevPoint
            </p>
            <p className="my-3 p-2 is-size-4 has-text-centered">
              Our experienced developers specialize in latest frameworks and technologies, and have
              a track record of successful project deliveries across multiple
              domains in the software industry
            </p>
            <p className="my-3 p-2 is-size-4 has-text-centered">
              Our platform allows you to meet our developers individually for a
              free online consultation and build rapport for future collaboration
            </p>
          </div>
          <div className="columns is-centered my-6 mx-5">
            <div className="column has-text-centered is-flex is-justify-content-center is-align-items-center">
              <Link to="/sign_up" className="button is-success is-large is-uppercase">Sign Up Now</Link>
            </div>
            <div className="column has-text-centered is-flex is-justify-content-center is-align-items-center">
              <p className="px-4 has-text-white has-text-weight-bold">or</p>
            </div>
            <div className="column has-text-centered is-flex is-justify-content-center is-align-items-center">
              <Link to="/login" className="button is-light is-small has-text-dark has-text-weight-bold">Log into your account</Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default LandingPage;
