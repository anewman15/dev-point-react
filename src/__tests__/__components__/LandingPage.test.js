import { unmountComponentAtNode } from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import LandingPage from '../../components/containers/LandingPage';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('LandingPage', () => {
  it('renders the LandingPage component to the dom', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <LandingPage />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
