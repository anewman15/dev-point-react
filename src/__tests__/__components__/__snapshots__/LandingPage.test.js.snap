// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders the LandingPage component to the dom 1`] = `
<div
  className="bg-code"
>
  <div
    className="columns is-centered"
  >
    <div
      className="my-6 column is-four-fifths"
    >
      <div
        className="p-6 bd-filter border-warning has-text-white"
      >
        <p
          className="is-size-2 has-text-centered"
        >
          Welcome to DevPoint
        </p>
        <p
          className="my-3 p-2 is-size-4 has-text-centered"
        >
          Our experienced developers specialize in latest frameworks and technologies, and have a track record of successful project deliveries across multiple domains in the software industry
        </p>
        <p
          className="my-3 p-2 is-size-4 has-text-centered"
        >
          Our platform allows you to meet our developers individually for a free online consultation and build rapport for future collaboration
        </p>
      </div>
      <div
        className="my-6 mx-5 has-text-centered is-flex is-justify-content-center is-align-items-center"
      >
        <a
          className="button is-success is-large is-uppercase"
          href="/sign_up"
          onClick={[Function]}
        >
          Sign Up Now
        </a>
        <p
          className="px-4 has-text-white has-text-weight-bold"
        >
          or
        </p>
        <a
          className="button is-light is-small has-text-dark has-text-weight-bold"
          href="/login"
          onClick={[Function]}
        >
          Log into your account
        </a>
      </div>
    </div>
  </div>
</div>
`;
